### ISSUE TITLE
Mongoose 4

### ISSUE THREAD

**Issue by rboorman** [baugarten/node-restful#83]
Got an error on dependencies when installing mongoose via npm.  node-restful apparently is looking for mongoose 3 and 4 is the latest.  Do we update the package.son to 4?  Have you tested node-restful with mongoose 4?  Thank you

**Comment by learningpro** [comment]
I found the same problem when I want to install using "npm install --save node-restful", it shows
```bash
    npm ERR! peerinvalid The package mongoose does not satisfy its siblings' peerDependencies     requirements!
    npm ERR! peerinvalid Peer node-restful@0.1.18 wants mongoose@~3
```

**Comment by Eric-Vandenberg** [comment]
I too am having this problem.  I ended up npm installing node-restful first in order for it to automatically install mongoose version 3.9.7 to avoid getting that error  HOWEVER this old build is unstable for production.  Now when running my server I see:

##############################################################
#
#   !!! MONGOOSE WARNING !!!
#
#   This is an UNSTABLE release of Mongoose.
#   Unstable releases are available for preview/testing only.
#   DO NOT run this in production.
#
##############################################################

**Comment by razvanilin** [comment]
If you install node-restful before mongoose you will get the latest mongoose 3 version which is the one mentioned above, 3.9.7. This release is unstable and introduced an annoying bug which gave me a hard time recently.
For the time being, I think the best approach is to use mongoose@3.8.23

**Comment by akolpakov** [comment]
+1 to support mongoose 4

**Comment by baugarten** [comment]
Ok, I'm going to look into what the breaking changes are. We don't use very complex APIs, so I have to imagine it will be a clean upgrade

**Comment by bmiller59** [comment]
+1

**Comment by romain10009** [comment]
+1

**Comment by adriangordon1231** [comment]
+1 +1 +1 +1 +1

**Comment by carloscant** [comment]
+1

**Comment by puradawid** [comment]
+1, we need that support.

**Comment by ThomasBormans** [comment]
+1

**Comment by baugarten** [comment]
Done. Thanks @maxaille and @rlebron88!

**Comment by baugarten** [comment]
This is available in version 0.2.1
